test/test_cp2k_mm_mock.py

```diff
@@ -0,0 +1,141 @@
+"""Mock CP2K funcionality."""
+
+import os
+import shutil
+
+import numpy as np
+from assertionlib import assertion
+from pytest_mock import mocker, MockFixture
+from scm.plams import Molecule
+
+from qmflows import Settings, cp2k_mm, singlepoint, geometry, freq, md
+from qmflows.utils import InitRestart
+from qmflows.packages.cp2k_mm import CP2KMM_Result
+from qmflows.test_utils import get_mm_settings, PATH, PATH_MOLECULES
+
+MOL = Molecule(PATH_MOLECULES / 'Cd68Cl26Se55__26_acetate.xyz')
+WORKDIR = PATH / 'output_cp2k_mm'
+
+#: Example input Settings for CP2K mm calculations.
+SETTINGS: Settings = get_mm_settings()
+
+# Ensure that plams.config is populated with a JobManager
+with InitRestart(PATH, 'tmp'):
+    pass
+if os.path.isdir(PATH / 'tmp'):
+    shutil.rmtree(PATH / 'tmp')
+
+
+def overlap_coords(xyz1: np.ndarray, xyz2: np.ndarray) -> np.ndarray:
+    """Rotate *xyz1* such that it overlaps with *xyz2* using the Kabsch algorithm."""
+    xyz1 = np.asarray(xyz1, dtype=float)
+    xyz2 = np.asarray(xyz2, dtype=float)
+
+    # Peform a singular value decomposition on the covariance matrix
+    H = xyz1.T @ xyz2
+    U, _, Vt = np.linalg.svd(H)
+    V, Ut = Vt.T, U.T
+
+    # Construct the rotation matrix
+    rotmat = np.eye(3)
+    rotmat[2, 2] = np.linalg.det(V @ Ut)
+    rotmat = V @ rotmat @ Ut
+
+    return xyz1 @ rotmat.T
+
+
+def mock_runner(mocker_instance: MockFixture,
+                settings: Settings = SETTINGS,
+                jobname: str = 'job') -> CP2KMM_Result:
+    """Create a Result instance using a mocked runner."""
+    run_mocked = mocker_instance.patch("qmflows.run")
+
+    dill_path = WORKDIR / jobname / f"{jobname}.dill"
+    plams_dir = WORKDIR / jobname
+
+    run_mocked.return_value = CP2KMM_Result(
+        settings, MOL, jobname,
+        dill_path=dill_path,
+        plams_dir=plams_dir,
+        work_dir=WORKDIR,
+        status='successful'
+    )
+    return run_mocked
+
+
+def test_cp2k_singlepoint_mock(mocker) -> None:
+    """Mock a call to CP2K."""
+    s = SETTINGS.copy()
```

Comment:
Why copy and not just a plain instantiation?

Reply:
Because I'm paranoid about passing mutable global constants without copying them.

Reply:
It is very good to have paranoid people building software that is the sanest mindset when programming! My objection about that line of code is that if you need to call a `@staticmethod` instead of the plain constructor (e.g. `Settings()`) it means that the constructor is somehow broken and that the `staticmethod` should be the default

Reply:
Point of clarify:
``SETTINGS`` is a ``Settings`` instance, not the ``Settings`` class itself.